Hints.py

```diff
@@ -1024,10 +1060,41 @@ def buildWorldGossipHints(spoiler, world, checkedLocations=None):
 
     hint_types, hint_prob = zip(*hint_dist.items())
     hint_prob, _ = zip(*hint_prob)
+    
+    # Add required dual location hints, only if hint copies > 0
+    if hint_dist['dual_always'][1] > 0:
+        alwaysDuals = getHintGroup('dual_always', world)
+        for hint in alwaysDuals:
+            dual = getDual(hint.name)
+            location = world.get_location(dual.firstLocation)
+            secondLocation = world.get_location(dual.secondLocation)
+            checkedAlwaysLocations.add(location.name)
+            checkedAlwaysLocations.add(secondLocation.name)
+            
+            if location.item.name in bingoBottlesForHints and world.settings.hint_dist == 'bingo':
```

Comment:
Instead of repeating this here and in bingo processing, add a new function for maintainability.